edgelet/edgelet-http/src/lib.rs

```diff
@@ -186,6 +210,38 @@ impl HyperExt for Http {
                     .with_context(|_| ErrorKind::BindListener(BindListenerType::Address(addr)))?;
                 Incoming::Tcp(listener)
             }
+            #[cfg(unix)]
+            HTTPS_SCHEME => {
+                let addr = url
+                    .to_socket_addrs()
+                    .context(ErrorKind::InvalidUrl(url.to_string()))?
```

Comment:
Nit: Wish there was a way to avoid the alloc when calling `url.to_string()` when there's no error.

Reply:
AFAICT there really isn't a way to do this at this time.